src/game.cpp

```diff
@@ -2731,10 +2731,17 @@ void Game::playerAcceptTrade(uint32_t playerId)
 			if (tradePartnerRet == RETURNVALUE_NOERROR && playerRet == RETURNVALUE_NOERROR) {
 				tradePartnerRet = internalMoveItem(playerTradeItem->getParent(), tradePartner, INDEX_WHEREEVER, playerTradeItem, playerTradeItem->getItemCount(), nullptr, FLAG_IGNOREAUTOSTACK, nullptr, partnerTradeItem);
 				if (tradePartnerRet == RETURNVALUE_NOERROR) {
-					internalMoveItem(partnerTradeItem->getParent(), player, INDEX_WHEREEVER, partnerTradeItem, partnerTradeItem->getItemCount(), nullptr, FLAG_IGNOREAUTOSTACK);
-					playerTradeItem->onTradeEvent(ON_TRADE_TRANSFER, tradePartner);
-					partnerTradeItem->onTradeEvent(ON_TRADE_TRANSFER, player);
-					isSuccess = true;
+					// check again if we can move this item to player
+					playerRet = internalAddItem(player, partnerTradeItem, INDEX_WHEREEVER, 0, true);
```

Comment:
It's almost functional, there is still bug which depends on the last person who pressed 'Accept' button. This means, you also need to check again for **tradePartnerRet**

Reply:
at this point player item is already in partner inventory, so all that is left is to check if partner item can be moved to player. Could you elaborate futher cause maybe I dont understand what you mean

Reply:
When I was working on this function yesterday I was trying different solutions, including something similar to yours. There was some sort of inconsistency with who is **player** and **tradePartner**. After several minutes of testing, it became clear that the **player** and **tradePartner** change based on who finalised the trade. By 'finalise', we mean the player who pressed 'Accept' button last. This comes down to what is written in the code right now.

Consider these two scenarios:

1.
Player 1 initialised trade with Player 2
Player 1 finalised the trade by pressing Accept button last.
This results in bypassing your code.

2.
Player 1 initialised trade with Player 2
Player 2 finalised the trade by pressing Accept button last.
This results in correct execution of your code.